gammapy/spectrum/models.py

```diff
@@ -483,6 +522,38 @@ def energy_flux(self, emin, emax):
             lower = (emin / pars['reference'].quantity) ** val
             return prefactor * (upper - lower)
 
+    def energy_flux_error(self, emin, emax, **kwargs):
+        r"""
+        Compute energy flux in given energy range analytically with error propagation.
+
+        Parameters
+        ----------
+        emin : `~astropy.units.Quantity`
+            Lower bound of integration range.
+        emax : `~astropy.units.Quantity`
+            Upper bound of integration range
+        """
+        emin = self._convert_energy(emin)
+        emax = self._convert_energy(emax)
+
+        unit = self.energy_flux(emin, emax, **kwargs).unit
+        upars = self.parameters._ufloats
+
+        val = -1 * upars['index'] + 2
+
+        if np.isclose(val.n, 0):
```

Comment:
Please use `.nominal_value` and `.std_dev` everywhere.
The `.n` and `.s` is really cryptic for people like me that aren't using `uncertainties` much.